src/conversions.ts

```diff
@@ -86,26 +86,29 @@ const convertAngleValueNode =
  * Returns null if the spherical string cannot be parsed.
  */
 export const deserializeSpherical =
-    (sphericalString: string): [number, number, number|string]|null => {
-      try {
-        const sphericalValueNodes = parseValues(sphericalString);
-
-        if (sphericalValueNodes.length === 3) {
-          const [thetaNode, phiNode, radiusNode] = sphericalValueNodes;
-
-          const theta = convertAngleValueNode(thetaNode);
-          const phi = convertAngleValueNode(phiNode);
-          const radius = radiusNode.value === 'auto' ?
-              'auto' :
-              lengthValueNodeToMeters(radiusNode);
-
-          return [theta, phi, radius];
-        }
-      } catch (_error) {
-      }
-
-      return null;
-    };
+    (sphericalString: string,
+     defaultValues: [number, number, number|null, number|null]):
+        [number, number, number|null, number|null] => {
```

Comment:
Is the returned factor ever considered outside of this helper, aside from immediately interpreting the result? Perhaps we can pre-apply the factor to the returned radius? That way we can stick to returning three not-null numbers always.

Also, let's document the fourth default in the comment above.

Reply:
Unfortunately I couldn't find a way to do that. The two numbers need to be used in different ways (only one of them is ever valid). I toyed with returning something that said how to use that number instead. I could use the sign as a flag, but figured that's pretty opaque. Do you have a better way?